ansible/modules/hashivault/hashivault_k8s_auth_config.py

```diff
@@ -72,15 +74,18 @@ def hashivault_k8s_auth_config(module):
     desired_state['pem_keys'] = params.get('pem_keys')
     desired_state['mount_point'] = mount_point
 
-    result = client.sys.list_auth_methods()
-    backends = result.get('data', result)
-    if (mount_point + "/") not in backends:
-        return {'failed': True, 'msg': 'auth method is not enabled', 'rc': 1, 'changed': False}
+    keys_updated = []
+    try:
+        current_state = client.auth.kubernetes.read_config(mount_point=mount_point)
+        keys_updated = get_keys_updated(current_state, desired_state)
+        if not keys_updated:
+            return {'changed': False}
+    except InvalidPath:
+        pass
 
     if not module.check_mode:
         client.auth.kubernetes.configure(**desired_state)
-
-    return {'changed': True}
+    return {'changed': True, 'keys_updated': keys_updated}
```

Comment:
nit: In the event of `InvalidPath` this will return `keys_updated==[]` where you probably want `keys_updated==desired_state.keys()`?

Reply:
Actually i'm not even sure how likely this is because if the auth mechanism is mounted then the read should succeed? Doesn't really matter.

Reply:
Just did it anyway https://github.com/TerryHowe/ansible-modules-hashivault/commit/dded693277c9028e6aa7fc2f4fc991258ead9601

Reply:
available in 4.5.1